packages/block-library/src/file/index.php

```diff
@@ -59,6 +38,19 @@ static function ( $matches ) {
 		$processor->next_tag( 'object' );
```

Comment:
generally it's nice to see official code check the response to `next_tag()` to ensure the tags are there. it could be the case that something changed it along the way. the writes will be benign, but the updates won't happen and we won't be the wiser.

this goes for the first tag above too.

```php
$processor = new WP_HTML_Tag_Processor( $content );
if ( $processor->next_tag() ) {
	$processor->set_attribute( 'data-wp-interactive', 'core/file' );
}

// If OBJECT is the first element, this could pass immediately.
while ( 'OBJECT' !== $processor->get_tag() && $processor->next_tag( 'object' ) ) {
	continue;
}

$filename = null;
if ( 'OBJECT' === $processor->get_tag() ) {
	$filename = $processor->get_attribute( 'aria-label' );
}

$has_filename = is_string( $filename ) && ! empty( $filename ) && 'PDF embed' !== $filename;
```

Reply:
Thanks for the feedback, @dmsnell!

Do we still want to do this when we know the `$content` markup?

I don't mind working on a follow-up PR; I'm unsure about current HTML API best practices.

Reply:
@dmsnell, that makes sense. I will follow-up with the File block code updates.

Question: HTML API still doesn't allow text content update/replacement, correct? See: https://github.com/WordPress/gutenberg/issues/60461#issuecomment-2038988286

Reply:
This is correct. It's actually quite easy to add in a subclass, but I would prefer we don't ship any such examples in Core.

One of the challenges in the Tag Processor is that it would be easy to assume that a `#text` node is a tag's inner text, when in fact it may only be one small piece of it.

If you want to explore that, let's chat in Slack on the `#core-html-api` channel. Again, I would urge against its use in Core before Core has its own method to replace that content.